src/Type/Regex/RegexGroupParser.php

```diff
@@ -64,20 +65,25 @@ public function parseGroups(string $regex): ?array
 			return null;
 		}
 
-		$rawRegex = $this->regexExpressionHelper->removeDelimitersAndModifiers($regex);
-		try {
-			$ast = self::$parser->parse($rawRegex);
-		} catch (Exception) {
-			return null;
-		}
-
 		$modifiers = $this->regexExpressionHelper->getPatternModifiers($regex) ?? '';
 		foreach (self::NOT_SUPPORTED_MODIFIERS as $notSupportedModifier) {
 			if (str_contains($modifiers, $notSupportedModifier)) {
 				return null;
 			}
 		}
 
+		if (str_contains($modifiers, 'x')) {
+			// in freespacing mode the # character starts a comment and runs until the end of the line
+			$regex = preg_replace('/[^?]#.*/', '', $regex) ?? '';
```

Comment:
This will also get rid of the character before # due to `[^?]`. I realize in most sane cases this will be a space and thus it's harmless, but IMO this should rather be a lookbehind like `(?<!\?)`

Reply:
will look into it, thanks for the heads up

Reply:
fix in https://github.com/phpstan/phpstan-src/pull/3739